python/sanlock.c

```diff
@@ -1395,15 +1438,20 @@ py_killpath(PyObject *self __unused, PyObject *args, PyObject *keywds)
 
     /* configure killpath (gil disabled) */
     Py_BEGIN_ALLOW_THREADS
-    rv = sanlock_killpath(sanlockfd, 0, path, kpargs);
+    rv = sanlock_killpath(sanlockfd, 0, PyBytes_AsString(path), kpargs);
     Py_END_ALLOW_THREADS
+    Py_DECREF(path);
```

Comment:
Decrefing path both on success and error path means we need to switch to single return at the end
and do the cleanup always in the end.